Packages/com.llealloo.audiolink/Runtime/Scripts/AudioLinkController.cs

```diff
@@ -161,14 +161,14 @@ public void UpdateSettings()
                 _threshold3Rect.anchorMin = anchor3;
                 // threshold3Rect.anchorMax is a constant value. Skip
 
-                audioSpectrumDisplay.SetFloat("_X0", x0Slider.value);
-                audioSpectrumDisplay.SetFloat("_X1", x1Slider.value);
-                audioSpectrumDisplay.SetFloat("_X2", x2Slider.value);
-                audioSpectrumDisplay.SetFloat("_X3", x3Slider.value);
-                audioSpectrumDisplay.SetFloat("_Threshold0", threshold0Slider.value);
-                audioSpectrumDisplay.SetFloat("_Threshold1", threshold1Slider.value);
-                audioSpectrumDisplay.SetFloat("_Threshold2", threshold2Slider.value);
-                audioSpectrumDisplay.SetFloat("_Threshold3", threshold3Slider.value);
+                audioSpectrumDisplay.SetFloat(audioLink._X0, x0Slider.value);
```

Comment:
Yes

Reply:
Unresolving so we have somewhere to talk about this: I suppose we should check that this is actually worth it given the overhead (probably?) of getting a field from another udon behavior. I don't know, maybe you could even check by just compiling and looking at the uasm - it might be obvious

Reply:
it's one extern 
`EXTERN,"VRCUdonCommonInterfacesIUdonEventReceiver.__GetProgramVariable__SystemString__SystemObject"`
plus one extern to cast it `EXTERN, "SystemConvert.__ToInt32__SystemObject__SystemInt32"`

Reply:
Ok I've decided to have each Behaviour keep track of it's own IDs, unless you have any objections I will go ahead with that

Reply:
just make the fields not internal anymore then

Reply:
already done, just haven't pushed yet